src/akushon/action/node/action_node.cpp

```diff
@@ -88,14 +90,41 @@ ActionNode::ActionNode(
   brake_action_subscriber = node->create_subscription<Empty>(
     brake_action_topic(), 10,
     [this](std::shared_ptr<Empty> message) {this->action_manager->brake();});
+  
+  ball_subscriber = node->create_subscription<Float64>(
+    ball_topic(), 10, [this](const std::shared_ptr<Float64> message) {
+      ball_pos = message->data;
+    });
 }
 
 bool ActionNode::start(const std::string & action_name)
 {
   Pose pose = this->initial_pose;
 
   if (!pose.get_joints().empty()) {
-    action_manager->start(action_name, pose);
+    if (action_name == akushon::ActionName::LEFT_KICK_WIDE || action_name == akushon::ActionName::RIGHT_KICK_WIDE)
+    {
+      std::size_t pos = action_name.find("_wide");
+      std::string source_action;
+      bool right = false;
+    
+      if (pos != std::string::npos) {
+        source_action = action_name.substr(0, pos);
+      } else {
+        source_action = action_name;
+      }
+
+      if (source_action == "right_kick") {
+        right = true;
+      }
+      if (action_manager->using_dynamic_kick)
+        action_manager->start(source_action, action_name, pose, ball_pos, action_manager->right_map_x_min, action_manager->right_map_x_max, action_manager->left_map_x_min, action_manager->left_map_x_max, right);
```

Comment:
No need to have a `right` variable if you can do this:
```suggestion
        action_manager->start(source_action, action_name, pose, ball_pos, action_manager->right_map_x_min, action_manager->right_map_x_max, action_manager->left_map_x_min, action_manager->left_map_x_max, source_action == "right_kick");
```